setup.py

```diff
@@ -47,5 +47,5 @@
       keywords=["string parsing", "string formatting", "pytroll"],
       zip_safe=False,
       install_requires=[],
-      tests_require=['pytest']
+      tests_require=['pytest', 'parameterized']
```

Comment:
Does `parameterized` have some benefit over `@pytest.mark.parametrize`?

Reply:
I tried `@pytest.mark.parametrize` first and it didn't work for me (the test failed complaining about missing argument). That's the only reason why I've used `parameterized`.

Reply:
Ok. I'd prefer not to add another dependency when it should be have the same. My guess is that you tried to use the decorator on a unittest TestCase subclass. That won't work. You'd need to extract the test methods to a separate class or function. Or you could rewrite the class so it doesn't have a subclass, but you'd have to replace all `self.assertX(...)` usages with `assert ...`.

Reply:
Yes. Basically, I want it to pass the same tests as the regular `compose`, so I just added a parameter to the existing test `test_compose` to test both functions. And that test was `unittest.TestCase`.

But sure, I can refactor the test so that it's in another class. I wasn't sure it this was OK, that's why I didn't do this originally.

Reply:
Yeah, thanks for being cautious, but in this case just rip it apart and make it work. We're slowly replacing unittest classes in pytroll packages with plain classes any way. At least, we're doing it as needed. This is one of those cases.